test/server-spec.js

```diff
@@ -0,0 +1,54 @@
+// Tests must be run while server is offline
+// Expected database situation is after initial seed with 10000000 data points
+
+const assert = require('assert');
+const expect = require('chai').expect;
+const supertest = require('supertest');
+
+// Set up mongoose with promises since default promise library is deprecated
+const mongoose = require('mongoose');
+mongoose.Promise = require('bluebird');
+
+const db = require('../database/AbandonedTotal.js');
+mongoose.connect('mongodb://localhost/abandonment_calculator', { useMongoClient: true });
+const app = require('../server/index.js');
+
+describe('Calculation Process', () => {
+  after(function(done) {
+    this.timeout(0);
+    console.log('Resetting database...');
+    db.AbandonedTotal.remove({viewInstanceId: 92374985})
+      .then(() => {
+        console.log('All tests done!');
+        done();
+      });
+  });
+  describe('View Route', () => {
+    it('Responds with 200', (done) => {
+      supertest(app)
+        .post('/view')
+        .set('Content-Type', 'application/json')
+        .send('{"events":[{"videoId":12345,"viewInstanceId":92374985,"eventAction":"PLAY","event_timestamp":"2017-12-08 23:28:31+00:00"},{"videoId":12345,"viewInstanceId":92374985,"eventAction":"PAUSE","event_timestamp":"2017-12-08 23:33:31+00:00"},{"videoId":12345,"viewInstanceId":92374985,"eventAction":"END","event_timestamp":"2017-12-08 23:38:31+00:00"}]}')
+        .expect(200, done);
+    });
+  });
+  describe('Calculation and Database Writing', (done) => {
```

Comment:
This test seems too specific, it will only work in the case of ten million entries? You may want to get the count of items, add one, then check the new count is +1 to make this test work in all cases.